solutions/kubernetes/cluster-functions.sh

```diff
@@ -283,8 +283,12 @@ function setup_primary_node(){
             CALICO_PLUGIN_MAJOR_VERSION=$(echo $CALICO_PLUGIN_VERSION | awk -F[.] '{print $1"."$2}')
             curl https://docs.projectcalico.org/archive/$CALICO_PLUGIN_MAJOR_VERSION/manifests/calico.yaml -o calico-$CALICO_PLUGIN_VERSION.yaml || throw $Exception    
         fi
-        # Setup IP_AUTODETECTION_METHOD for determining calico network.
-        # sed -i '/# Auto-detect the BGP IP address./i \            - name: IP_AUTODETECTION_METHOD\n              value: "interface=eth-0"' calico-$CALICO_PLUGIN_VERSION.yaml
+
+        IS_VM=$(systemd-detect-virt -v)
+        if [ "$IS_VM" == "none" ]; then
+            # Setup IP_AUTODETECTION_METHOD for determining calico network.
+            sed -i '/# Auto-detect the BGP IP address./i \            - name: IP_AUTODETECTION_METHOD\n              value: "interface=eno1"' calico-$CALICO_PLUGIN_VERSION.yaml
```

Comment:
Can we avoid using hardcoded `eno1`

Reply:
@shailesh-vaidya, if not `eno1` hardcoded, we need to get interface name using ip which starts with 10. is that fine?